modules/selection/src/main/kotlin/com.r3.corda.lib.tokens.selection/memory/services/VaultWatcherService.kt

```diff
@@ -279,8 +300,14 @@ class VaultWatcherService(private val tokenObserver: TokenObserver,
         indexViewCreationLock.read {
             for (stateAndRef in stateAndRefs) {
                 val existingMark = __backingMap.remove(stateAndRef)
-                existingMark
-                        ?: LOG.warn("Attempted to remove existing token ${stateAndRef.ref}, but it was not found this suggests incorrect vault behaviours")
+                if (existingMark == null) {
+                    if (tokenLoadingFinished) {
```

Comment:
`!tokenLoadingStarted || tokenLoadingFinished`

For true correctness.  There's scenarios in the tests where it never starts loading, but you probably want the warning. (Via the other constructor).